utils/DeployUtils.ps1

```diff
@@ -161,63 +161,141 @@ function ConfigureScubaGearModule{
     return $null -ne $Result
 }
 
-function SignScubaGearModule{
+function CreateFileList{
     <#
-        .NOTES
-        Internal helper function
+    .NOTES
+    Internal function 
+    #>
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $SourcePath,
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyCollection()]
+        [array]
+        $LiteralFilePaths = @(),
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyCollection()]
+        [array]
+        $Extensions = @()
+    )
+
+    $FileNames = @()
+
+    if ($Extensions.Count -gt 0){
+        $FileNames += Get-ChildItem -Recurse -Path $SourcePath -Include $Extensions
+    }
+    
+    Write-Debug "Found $($FileNames.Count) files to sign" 
+
+    $FileList = New-TemporaryFile
+    $FileNames.FullName | Out-File -FilePath $($FileList.FullName) -Encoding utf8 -Force
+    Write-Debug "Files: $(Get-Content $FileList)"
+    return $FileList.FullName
+}
+
+function CallAzureSignTool{
+    <#
+    .NOTES
+    Internal function 
     #>
     param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'https'})]
+        [System.Uri]
+        $AzureKeyVaultUrl,
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $CertificateName,
         [Parameter(Mandatory=$true)]
         [ValidateScript({Test-Path -Path $_})]
         $ModulePath,
         [Parameter(Mandatory=$false)]
-        [scriptblock]
-        $GetCertificate = $function:GetCodeSigningCertificate,
-        [Parameter(Mandatory=$false)]
         [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'http','https'})]
         $TimeStampServer = 'http://timestamp.digicert.com',
-        [Parameter(Mandatory=$false)]
-        [ValidateSet('SHA256')]
-        [string]
-        $HashAlgorithm = 'SHA256'
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_ -PathType Leaf})]
+        $FileList
     )
-    #TODO: Add code signing
-    $CatalogFileName = 'ScubaGear.cat'
-    $CatalogPath = Join-Path -Path $ModulePath -ChildPath $CatalogFileName
-    $Cert = Invoke-Command $GetCertificate
 
-    # Digitally sign scripts, manifest, and modules
-    Get-ChildItem $ModulePath -Include *.psd1,*psm1,*.ps1 -Recurse |
-    Set-AuthenticodeSignature -Certificate $Cert -TimeStampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+    $SignArguments = @(
```

Comment:
Do we know if Azure Sign Tool rate limits its calls to the timestamp service at all?  Have we seen any flakiness or issues in the testing?  Assuming we haven't, I think this is fine but I was curious our level of confidence.

Reply:
Ok, we will see.